src/datastore/StringCrypto.ts

```diff
@@ -48,19 +66,66 @@ export class StringCrypto {
     }
 
     public encrypt(plaintext: string): string {
-        const salt = crypto.randomBytes(16).toString('base64');
-        return crypto.publicEncrypt(
+        const salt = randomBytes(SALT_BYTE_LENGTH).toString(SALT_ENCODING);
+        return publicEncrypt(
             this.privateKey,
             Buffer.from(salt + ' ' + plaintext)
-        ).toString('base64');
+        ).toString(ENCRYPTED_ENCODING);
     }
 
     public decrypt(encryptedString: string): string {
-        const decryptedPass = crypto.privateDecrypt(
+        const decryptedPass = privateDecrypt(
             this.privateKey,
-            Buffer.from(encryptedString, 'base64')
+            Buffer.from(encryptedString, ENCRYPTED_ENCODING)
         ).toString();
         // Extract the password by removing the prefixed salt and seperating space
-        return decryptedPass.split(' ')[1];
+        return decryptedPass.slice(SALT_STRING_LENGTH + 1);
+    }
+
+    public async encryptLargeString(plaintext: string): Promise<string> {
+        const password = randomBytes(32).toString(ENCRYPTED_ENCODING);
+        const key = await scrypt(password, 'salt', 32) as Buffer;
+        const iv = randomBytes(16);
+
+        const cipher = createCipheriv(algorithm, key, iv);
+        cipher.setEncoding(ENCRYPTED_ENCODING);
+        let encrypted = '';
+
+        // Large strings are encrypted as 'lg:encrypt($key_$iv):$encrypted_block' where the key_iv is further
+        // encrypted by the root private key.
+        const secret = this.encrypt(`${key.toString(ENCRYPTED_ENCODING)}_${iv.toString(ENCRYPTED_ENCODING)}`);
+        const streamPromise = new Promise<string>((resolve, reject) => {
+            cipher.on('error', (err) => reject(err));
+            cipher.on('end', () => resolve(
+                `lg:${secret}:${encrypted}`
+            ));
+        });
+
+        cipher.on('data', (chunk) => { encrypted += chunk });
+        cipher.write(plaintext);
+        cipher.end();
+        return streamPromise;
```

Comment:
Since the data to be written into the cipher is known to be just the provided `plaintext`, it'd be more direct to use `cipher.update` and `cipher.final` instead of bothering with stream callbacks.